ocp_addons_operators_cli/utils/general.py

```diff
@@ -89,3 +96,40 @@ def get_iib_dict():
         _iib_dict = extract_iibs_from_json(ocp_version=ocp_version, job_name=job_name)
 
     return _iib_dict
+
+
+def collect_must_gather(
+    must_gather_output_dir, kubeconfig_path, cluster_name, product_name
+):
+    try:
+        target_dir = os.path.join(
+            must_gather_output_dir, "must-gather", cluster_name, product_name
+        )
+    except Exception as ex:
```

Comment:
The error seems unrelated (you try to get target dir, not exec must gather here)
also this will never fail, it is just join()

Reply:
I took help from the `openshift-cli-installer` [code](https://github.com/RedHatQE/openshift-cli-installer/blob/main/openshift_cli_installer/libs/clusters/ocp_cluster.py#L245), now I understand it doesnot check the path, instead function will create this path itself.